iati/validator.py

```diff
@@ -122,7 +123,7 @@ def contains_error_called(self, err_name):
             bool: Whether there is an error or warning with the specified name within the log.
 
         """
-        errors_with_name = [err for err in self._values if err.name == err_name]
+        errors_with_name = self.get_errors_or_warnings_by_name(err_name)
```

Comment:
Need to think about how errors and warnings are treated differently, and how this impacts function naming - at the moment the name of this function only specifies `error`, though functionally returns both `errors` and `warnings`.

Probably needs more design work.

Suggest adding a `Warning` and `TODO` to the class saying that it is highly likely that the function names will change, while merely making things consistent at present.

Reply:
Added in 578009c379a4d0a1675854d0b5ed6bee7171c86e